### PR TITLE
Make sure vectorization and inlining is consistent in the reduction/norm kernels.

### PR DESCRIPTION
**Pull request by csarofeen**
Some scattered perf improvements, but on V100 it improved the TIMM VIT kernel5 micro benchmark by 4.45x.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp [hunk 1]
```diff
@@ -514,7 +536,7 @@ void multiReductionInliner(
         compute_from,
         compute_to,
         -1,
-        ComputeAtMode::MostInlined,
+        ComputeAtMode::BestEffort,
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp:539]: Not really clear what the motivation of this change is.

**Review by naoyam — APPROVED** [review]
Stamping as I don't see any problem, but not really clear why these changes matter.

File: torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp [hunk 2]
```diff
@@ -322,8 +325,19 @@ void multiReductionInliner(
       auto cached_output = cached_output_pair.first;
       auto output = cached_output_pair.second;
 
-      // If an output has multiple consumers don't process here, we want only
-      // terminating outputs
+      if (vectorize) {
+        if (std::find(
+                vecotrizable_inputs_outputs.begin(),
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp]: nit: misspelled `vectorizable_inputs_outputs`

File: torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp:539]: There shouldn't be any reductions present in that section so it shouldn't matter. Maybe we should use the least aggressive mode that fits the situation, but honestly don't know why I changed it. Shouldn't matter.

**Pull request merged**